extensions/sql-migration/src/wizard/sourceConfigurationPage.ts

```diff
@@ -0,0 +1,323 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as azdata from 'azdata';
+import { MigrationWizardPage } from '../models/migrationWizardPage';
+import { MigrationStateModel, StateChangeEvent } from '../models/stateMachine';
+import * as constants from '../constants/strings';
+import { IconPath, IconPathHelper } from '../constants/iconPathHelper';
+
+const headerLeft: azdata.CssStyles = {
+	'border': 'none',
+	'text-align': 'left',
+	'white-space': 'nowrap',
+	'text-overflow': 'ellipsis',
+	'overflow': 'hidden',
+	'border-bottom': '1px solid'
+};
+
+const styleLeft: azdata.CssStyles = {
+	'border': 'none',
+	'text-align': 'left',
+	'white-space': 'nowrap',
+	'text-overflow': 'ellipsis',
+	'overflow': 'hidden',
+};
+export class SourceConfigurationPage extends MigrationWizardPage {
+	private _view!: azdata.ModelView;
+	private _databaseSelectorTable!: azdata.DeclarativeTableComponent;
+	private _dbNames!: string[];
+	private _dbCount!: azdata.TextComponent;
+
+	constructor(wizard: azdata.window.Wizard, migrationStateModel: MigrationStateModel) {
+		super(wizard, azdata.window.createWizardPage(constants.SOURCE_CONFIGURATION, 'MigrationModePage'), migrationStateModel);
+	}
+
+	protected async registerContent(view: azdata.ModelView): Promise<void> {
+		this._view = view;
+
+		const flex = view.modelBuilder.flexContainer().withLayout({
+			flexFlow: 'row',
+			height: '100%',
+			width: '100%'
+		}).component();
+		flex.addItem(await this.createRootContainer(view), { flex: '1 1 auto' });
+
+		await view.initializeModel(flex);
+	}
+
+	public async onPageEnter(): Promise<void> {
+		this.wizard.registerNavigationValidator((pageChangeInfo) => {
+			return true;
+		});
+	}
+	public async onPageLeave(): Promise<void> {
+		this.migrationStateModel._databaseAssessment = this.selectedDbs();
+		this.wizard.registerNavigationValidator((pageChangeInfo) => {
+			return true;
+		});
+	}
+
+	protected async handleStateChange(e: StateChangeEvent): Promise<void> {
+	}
+
+
+	public async createRootContainer(view: azdata.ModelView): Promise<azdata.FlexContainer> {
+		const queryProvider = azdata.dataprotocol.getProvider<azdata.QueryProvider>((await this.migrationStateModel.getSourceConnectionProfile()).providerId, azdata.DataProviderType.QueryProvider);
+		const query =
+			`IF OBJECT_ID('tempdb.dbo.#space') IS NOT NULL
```

Comment:
I would recommend moving trying the queries on older sql servers to make sure we are back compat.